### ISSUE TITLE
UNSUPPORTED: Map literals are no longer supported in the template.

### ISSUE THREAD

**Issue by reducedcarpet** [dart-league/ng_bootstrap#133]
First off, I would like to apologise because I have no clue as to whether this issue to due to the version of AngularDart my GitLab CI is using. But I do get a lot of errors trying to build ng_bootstrap all of them to do with map literals: 

[SEVERE]angular:angular on package:ng_bootstrap/components/progress/progress.dart: Template parse errors:
line 9, column 15 of asset:ng_bootstrap/lib/components/progress/progress.dart: ParseErrorLevel.FATAL: Parser Error: UNSUPPORTED: Map literals are no longer supported in the template.
Move code that constructs or maintains Map instances inside of your @Component-annotated Dart class, or prefer syntax such as [class.active]="isActive" over [ngClass]="{'active': isActive}".  [{$implicit: percentage, value: value, max: max}] in null
  ╷
9 │               [ngTemplateOutletContext]="{$implicit: percentage, value: value, max: max}"></template>
  │               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵

This is pretty much the same as the 30 ish other errors that appear after it. 

I don't get this when I build it locally, and it doesn't effect manually deploying the code. But it does break the CI pipeline. 

So what I'm asking is, is this a known issue with some version of dart or angulardart? Do I have extreme linting on somehow?  Do I need to upgrade / downgrade some stuff? Was I only suppose to use this with Dart1?

**Comment by luisvt** [comment]
what version are you using?

**Comment by reducedcarpet** [comment]
Apparently the CI pipeline is using dart 2.6 (an webdev 2.6.3 I think) at least, on my home machine I am using dart 2.4.1 and webdev version is 2.5.0, whereas PUB is 2.4.1. I downgraded the CI pipeline to 2.4 and it no longer complains about the map literals. I hope that helps.

**Comment by luisvt** [comment]
sorry, what version of `ng_bootstrap`?

**Comment by reducedcarpet** [comment]
Oh my Bad, the one on pub.dev: 1.1.1

**Comment by luisvt** [comment]
try version [1.2.0-alpha.1](https://pub.dev/packages/ng_bootstrap/versions/1.2.0-alpha.1)

**Comment by reducedcarpet** [comment]
Yeah I have now, and unfortunately it still had the same errors and refused to build. I tried it on Dart 2.5 as well and it was the same. (with the alpha version) so I still have to use dart 2.4

**Comment by luisvt** [comment]
ok, let me check then

**Comment by konei** [comment]
I ran into this with angular 6.0.0-alpha.1
Knocking it down to alpha only fixed it:

dependencies:
  angular: 6.0.0-alpha

I don't know the implications for future releases.

**Comment by zarisi** [comment]
Fix for all the components made in #134
